rllib/agents/dqn/dqn.py

```diff
@@ -202,64 +241,39 @@ def get_initial_state(config):
     }
 
 
-def make_exploration_schedule(config, worker_index):
-    # Use either a different `eps` per worker, or a linear schedule.
-    if config["per_worker_exploration"]:
-        assert config["num_workers"] > 1, \
-            "This requires multiple workers"
-        if worker_index >= 0:
-            # Exploration constants from the Ape-X paper
-            exponent = (
-                1 + worker_index / float(config["num_workers"] - 1) * 7)
-            return ConstantSchedule(0.4**exponent)
-        else:
-            # local ev should have zero exploration so that eval rollouts
-            # run properly
-            return ConstantSchedule(0.0)
-
-    return PiecewiseSchedule(
-        endpoints=[
-            (0, config["exploration_initial_eps"]),
-            (int(config["exploration_fraction"] *
-                 config["schedule_max_timesteps"]),
-             config["exploration_final_eps"]),
-        ],
-        outside_value=config["exploration_final_eps"])
-
-
-def setup_exploration(trainer):
-    trainer.exploration0 = make_exploration_schedule(trainer.config, -1)
-    trainer.explorations = [
-        make_exploration_schedule(trainer.config, i)
-        for i in range(trainer.config["num_workers"])
-    ]
-
-
-def update_worker_explorations(trainer):
+# TODO(sven): Move this to generic Trainer/Policy. Every Algo should do this.
+def update_worker_exploration(trainer):
+    """Sets epsilon exploration values in all policies to updated values.
+
+    According to current time-step.
+
+    Args:
+        trainer (Trainer): The Trainer object for the DQN.
+    """
+    # Store some data for metrics after learning.
     global_timestep = trainer.optimizer.num_steps_sampled
-    exp_vals = [trainer.exploration0.value(global_timestep)]
-    trainer.workers.local_worker().foreach_trainable_policy(
-        lambda p, _: p.set_epsilon(exp_vals[0]))
-    for i, e in enumerate(trainer.workers.remote_workers()):
-        exp_val = trainer.explorations[i].value(global_timestep)
-        e.foreach_trainable_policy.remote(lambda p, _: p.set_epsilon(exp_val))
-        exp_vals.append(exp_val)
     trainer.train_start_timestep = global_timestep
-    trainer.cur_exp_vals = exp_vals
+
+    # Get all current exploration-infos (from Policies, which cache this info).
+    trainer.exploration_infos = trainer.workers.foreach_trainable_policy(
+        lambda p, _: p.last_exploration_info)
 
 
-def add_trainer_metrics(trainer, result):
+def after_train_result(trainer, result):
+    """Add some DQN specific metrics to results.
```

Comment:
style: the ending triple-quotes should be on the same line if it fits.

Reply:
done